### PR TITLE
feat: running server in the background

### LINKED ISSUE #960: Run the pnpm service in the background
Once #832 is done, we'd like an option to run the service in the background, automatically.

Possibly the https://github.com/IonicaBizau/diable can be used to achieve this

### PR DESCRIPTION
**Pull request by zkochan**
close #960

1. `pnpm server start` now supports the `--background` flag
2. `pnpm install` support the `--use-store-server` flag

### CODE REVIEW

File: src/cmd/server/start.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import logger from '@pnpm/logger'
 import {createServer} from '@pnpm/server'
+import Diable = require('diable')
```
Comment (IonicaBizau) [inline src/cmd/server/start.ts:3]: I think you can `import Diable from 'diable'` as well.

Comment (zkochan) [reply src/cmd/server/start.ts:3]: no, it is undefined when I import it via import from.

**Comment by zkochan** [comment]
@etamponi, do you like the naming? I think `--use-store-server` is fine but I think we might want to add some other options in the future. Some possible flags might:

* disallow installation using a store server
* allow installation only using a server but disallow creating one. That might be something Glitch would use

So maybe in this context, a different name would be better?

**Comment by zkochan** [comment]
yes, you are correct, it will wait for the newly created server to startup

and yes, all configs can be used via `.npmrc`.

**Pull request merged**